src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1015,3 +1057,960 @@ def upgrade_agents(cmd, client, resource_group_name, cluster_name, kube_config=N
+                                    summary='Exception while merging the kubeconfig file')
+            raise CLIError('Exception while merging the kubeconfig file.' + str(e))
+
+    current_context = addition.get('current-context', 'UNKNOWN')
+    msg = 'Merged "{}" as current context in {}'.format(current_context, existing_file)
+    print(msg)
+
+
+def handle_merge(existing, addition, key, replace):
+    if not addition[key]:
+        return
+    if existing[key] is None:
+        existing[key] = addition[key]
+        return
+
+    for i in addition[key]:
+        for j in existing[key]:
+            if i['name'] == j['name']:
+                if replace or i == j:
+                    existing[key].remove(j)
+                else:
+                    msg = 'A different object named {} already exists in your kubeconfig file.\nOverwrite?'
+                    overwrite = False
+                    try:
+                        overwrite = prompt_y_n(msg.format(i['name']))
+                    except NoTTYException:
+                        pass
+                    if overwrite:
+                        existing[key].remove(j)
+                    else:
+                        msg = 'A different object named {} already exists in {} in your kubeconfig file.'
+                        telemetry.set_exception(exception='A different object with same name exists in the kubeconfig file', fault_type=consts.Different_Object_With_Same_Name_Fault_Type,
+                                                summary=msg.format(i['name'], key))
+                        raise CLIError(msg.format(i['name'], key))
+        existing[key].append(i)
+
+
+def _resolve_service_principal(client, identifier):  # Uses service principal graph client
+    # todo: confirm with graph team that a service principal name must be unique
+    result = list(client.list(filter="servicePrincipalNames/any(c:c eq '{}')".format(identifier)))
+    if result:
+        return result[0].object_id
+    if utils.is_guid(identifier):
+        return identifier  # assume an object id
+    error = CLIError("Service principal '{}' doesn't exist".format(identifier))
+    error.status_code = 404  # Make sure CLI returns 3
+    raise error
+
+
+def client_side_proxy_wrapper(cmd,
+                              client,
+                              resource_group_name,
+                              cluster_name,
+                              token=None,
+                              path=os.path.join(os.path.expanduser('~'), '.kube', 'config'),
+                              context_name=None,
+                              api_server_port=consts.API_SERVER_PORT):
+
+    client_proxy_port = consts.CLIENT_PROXY_PORT
+    if token is not None:
+        telemetry.add_extension_event('connectedk8s', {'Context.Default.AzureCLI.IsAADEnabled': False})
+    else:
+        telemetry.add_extension_event('connectedk8s', {'Context.Default.AzureCLI.IsAADEnabled': True})
```

Comment:
>IsAADEnabled [](start = 81, length = 12)

I dont think we need this telemetry.. From the params we can identify how many times the CLI was invoked with service account token.

Reply:
yes you are right. Removed it.